src/output-json-ftp.c

```diff
@@ -107,40 +101,61 @@ static json_t *JsonFTPLogCommand(Flow *f, FTPTransaction *tx)
                 if (pos >= 3)  {
                     /* Gather the completion code if present */
                     if (isdigit(where[0]) && isdigit(where[1]) && isdigit(where[2])) {
-                        json_array_append_new(js_respcode_list,
-                                              JsonAddStringN((const char *)where, 3));
+                        char *s = BytesToString(where, 3);
+                        if (s != NULL) {
+                            jb_append_string(js_respcode_list, s);
```

Comment:
the goal of the jsonbuilder transition is to avoid creating temporary objects like this, but instead "stream" the whole object into a single `jb`.

Reply:
I modeled my implementation after that in `output-json-email-common.c`.

Note that I'm building 2 arrays as the response is processed and 2 arrays will be created during that. I'm not sure how the new interfaces support this?

Reply:
There isn't in some cases.  In DNS, to avoid looping throught the responses 2x, I had to use intermediate objects as well in `dns_log_json_answer`.